docs/guides/migrating-cra.md

```diff
@@ -0,0 +1,774 @@
+
+## Ensure your app uses React Router v6
+
+If you are using an older version of React Router, the first step is to upgrade to v6. Check out the [migration guide from v5 to v6](https://reactrouter.com/docs/en/v6/upgrading/v5) and our [backwards compatibility package](https://www.npmjs.com/package/react-router-dom-v5-compat) to upgrade your app quickly.
+
+## A note about non-standard imports
+
+CRA supports several non-standard file imports in your JavaScript/TypeScript modules. Since such imports don't actually work in plain JavaScript, they have to be transformed by a compiler into something that makes sense at runtime. This includes things like styles, images, and SVG files.
+
+Remix also supports some non-standard imports, but it's not a 1-to-1 mapping in terms of what we support or how we handle loading the file contents. Below is a non-exhaustive list of some of the differences you'll encounter in Remix.
+
+### Images and fonts
+
+CRA allows you to import image files, and result is a string representing the filepath of the image. This works the same way in Remix.
+
+```js
+import * as React from "react";
+
+import logo from "./logo.png";
+
+export function Logo() {
+  return <img src={logo} alt="My logo" />;
+}
+```
+
+This should work for fonts as well. In Remix, you'll generally import fonts in a route module and use them as the `href` of a `link` tag. [See our route loader docs for more information.](../api/conventions#loader)
+
+### SVG
+
+CRA allows you to import SVG files as React component. This is a common use case for SVG files, but it's not supported by default in Remix.
+
+```js
+import * as React from "react";
+
+// This will not work in Remix!
+import { ReactComponent as MyLogo } from "./logo.svg";
+
+export function Logo() {
+  return <MyLogo />;
+}
+```
+
+If you want to use SVG files as React components, you'll need to first create the components and import them directly. [React SVGR](https://react-svgr.com/) is a great toolset that can help you generate these components from the [command line](https://react-svgr.com/docs/cli/) or in an [online playground](https://react-svgr.com/playground/) if you prefer to copy and paste.
+
+### Style imports
+
+CRA supports importing styles in your components in many ways. While this is common practice in the React ecosystem, it's not supported the same way in Remix for a few different reasons.
+
+## Installing packages
+
+First, you'll need a few of our packages to build on Remix. Follow the instructions below, running all commands from the root of your CRA project.
+
+```shell
+npm install @remix-run/react @remix-run/node
+```
+
+Because CRA apps are client-rendered by default, you may not have a server yet. All Remix apps run on a server, so we'll install a few packages to create one quickly.
+
+```shell
+npm install @remix-run/express express@4 compression@1 morgan@1
```

Comment:
Is there a reason they can't just use remix-serve? That would simplify this guide considerably I think. If there's a change remix-serve needs to support CRA apps then maybe we should make that change.

Reply:
This guide was written to basically emulate our Express template, so I copied all of the scripts and dependencies directly from it. We could probably simplify it but I'd like to be as close to the template as possible, as we'll be linking out to it.

Reply:
@kentcdodds Discussed w/ @mjackson before I wrote this and he agreed the Express template would make for a better teaching opportunity. We could discuss again, just want to make sure we're all on the same page.